process/headerCheck/headerSignatureVerify.go

```diff
@@ -321,21 +324,37 @@ func (hsv *HeaderSigVerifier) VerifyHeaderWithProof(header data.HeaderHandler) e
 	return hsv.VerifyHeaderProof(prevProof)
 }
 
-func (hsv *HeaderSigVerifier) getHeaderForProof(proof data.HeaderProofHandler) (data.HeaderHandler, error) {
+func (hsv *HeaderSigVerifier) getHeaderForProofAtTransition(proof data.HeaderProofHandler) (data.HeaderHandler, error) {
 	headerUnit := dataRetriever.GetHeadersDataUnit(proof.GetHeaderShardId())
 	headersStorer, err := hsv.storageService.GetStorer(headerUnit)
 	if err != nil {
 		return nil, err
 	}
 
-	return process.GetHeader(proof.GetHeaderHash(), hsv.headersPool, headersStorer, hsv.marshalizer, proof.GetHeaderShardId())
+	var header data.HeaderHandler
+
+	for {
```

Comment:
maybe add a max number of retries here? this will wait for ever if the header is not received

Reply:
it will be anyway blocked if the header is not received